packages/commcare/src/Utils.js

```diff
@@ -1,10 +1,29 @@
+/* eslint-disable no-param-reassign */
 import { composeNextState } from '@openfn/language-common';
 import {
   request as commonRequest,
   makeBasicAuthHeader,
   logResponse,
 } from '@openfn/language-common/util';
 
+export const configureAuth = (auth, headers = {}) => {
+  const configKeys = Object.keys(auth);
+
+  if (!configKeys.includes('apiKey') && !configKeys.includes('password')) {
```

Comment:
I would restructure this if-else branch to be more positive. I tend to avoid `if(!x) {} else { }` branches, I much prefer to do `if(x) {} else { }`.

There are exceptions of course but I don't think this is one :)

So I'd write this as:

```
if ('apiKey' in auth) {
  headers = {
    Authorization: `ApiKey ${auth.username}:${auth.apiKey}`,
  };
} else if ('password' in auth) {
  headers = {
    ...makeBasicAuthHeader(auth.username, auth.password),
  };
} else {
  throw new Error('Invalid authorization credentials. Include an apiKey or password in state.configuration');
}
```

The error state is also a bit more comprehensive I think

Reply:
I'be not tested this, but I'd also be temped to use Object.assign  - I think it's a bit nicer than the spread?
```
if ('apiKey' in auth) {
  Object.assign(headers, {
    Authorization: `ApiKey ${auth.username}:${auth.apiKey}`,
  });
} else if ('password' in auth) {
  Object.assign(headers, makeBasicAuthHeader(auth.username, auth.password))
  };
} else {
  throw new Error('Invalid authorization credentials. Include an apiKey or password in state.configuration');
}
```